applications/openshift/kubelet/kubelet_enable_server_cert_rotation/rule.yml

```diff
@@ -33,7 +33,7 @@ ocil_clause: 'the kubelet cannot rotate server certificate'
 
 ocil: |-
     Run the following command on the kubelet node(s):
-    <pre>$ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep RotateKubeletServerCertificate; done</pre>
+    <pre>$ for node in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$node/proxy/configz | jq '.kubeletconfig.rotateCertificates' done</pre>
```

Comment:
@rhmdnd I am not quite sue, but looks like the rotateCertificates is for rule  ocp4-kubelet-enable-cert-rotation.
```
$ oc get rule ocp4-kubelet-enable-cert-rotation -o=jsonpath={.instructions}
Run the following command on the kubelet node(s):
$ sudo grep rotateCertificates /etc/kubernetes/kubelet.conf
The output should return nothing or true.
$(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc debug node/$node -- chroot /host grep rotateCertificates /etc/kubernetes/kubelet.conf; done
Warning: would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true, hostIPC=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/xiyuan-10-b414-q5dl4-master-0copenshift-qeinternal-debug ...
To use host binaries, run `chroot /host`
  "rotateCertificates": true,

Removing debug pod ...
Warning: would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true, hostIPC=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/xiyuan-10-b414-q5dl4-master-1copenshift-qeinternal-debug ...
To use host binaries, run `chroot /host`
^C
Removing debug pod ...
```

Reply:
https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#certificate-rotation

Reply:
@xiaojiey has a point here.

CIS 4.2.10 is about `.rotateCertificates` , and CIS 4.2.11 is about `.kubeletconfig.featureGates.RotateKubeletServerCertificate`

One thing I noticed is that in 4.13 `.kubeletconfig.featureGates.RotateKubeletServerCertificate` is set to true.
While in 4.14, `RotateKubeletServerCertificate` is missing from `.kubeletconfig.featureGates`.

Reply:
Ok - good catch. We might need to find another way to support this rule on OpenShift 4.14 then.

https://docs.openshift.com/container-platform/4.13/nodes/clusters/nodes-cluster-enabling-features.html

I don't see the `RotateKubeletServerCertificate` feature gate listed in the documentation.

Reply:
I think this relevant to why we see the feature gate in 4.13, but not 4.14.

https://github.com/openshift/machine-config-operator/pull/3688

I haven't found a way to determine which feature gates are enabled by default. Given that, maybe we only run this rule on 4.13 and older versions, and exclude it on 4.14.

Reply:
Actually - after reading https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#certificate-rotation it looks like the `RotateKubeletServerCertificate` feature was removed in favor of `rotateCertificates`.

Reply:
Nevermind - we should be using `serverTLSBootstrap` instead.